packages/draganddrop/src/components/DragAndDrop.tsx

```diff
@@ -0,0 +1,209 @@
+import {
+  DndContext,
+  DragEndEvent,
+  DragOverEvent,
+  DragOverlay,
+  DragStartEvent,
+  DropAnimation,
+  KeyboardSensor,
+  PointerSensor,
+  closestCorners,
+  defaultDropAnimation,
+  useSensor,
+  useSensors,
+} from '@dnd-kit/core';
+import { arrayMove, sortableKeyboardCoordinates } from '@dnd-kit/sortable';
+import { Table, TableBody, TableCell, TableHead, TableRow } from '@mui/material';
+import { ReactNode, useEffect, useState } from 'react';
+import BoardSection from './BoardSection';
+import DDItem from './DDItem';
+import { BoardSectionType, DDCategory, DDType } from './types';
+import { findBoardSectionContainer } from './utils/board';
+
+type DragAndDropProps<T extends DDType> = {
+  categories: DDCategory<string>[];
+  items: T[];
+  onChange: Function;
+  isCategory: (item: T, category: DDCategory<any>) => boolean;
+  cardRenderer?: (item: T) => ReactNode;
+};
+
+const DragAndDrop = <T extends DDType,>({ items, onChange, categories, isCategory, cardRenderer }: DragAndDropProps<T>) => {
+  const [boardSections, setBoardSections] = useState<BoardSectionType<T>>();
+
+  useEffect(() => {
+    const initialBoardSections = initializeBoard(items);
+    setBoardSections(initialBoardSections);
+  }, [items]);
+
+
+  const [activeItemId, setActiveItemId] = useState<null | number>(null);
+
+  const [changes, setChanges] = useState<Map<string, unknown>>(new Map());
+
+  const sensors = useSensors(
+    useSensor(PointerSensor),
```

Comment:
This will allow buttons to be used in cell renderers.
```
    useSensor(PointerSensor, {
      activationConstraint: {
        distance: 5, // Enable sort function when dragging 5px   💡 here!!!
      },
    }),
```